src/codeu/chat/client/commandline/Chat.java

```diff
@@ -107,11 +110,36 @@ public void invoke(Scanner args) {
         System.out.println("    Add a new user with the given name.");
         System.out.println("  u-sign-in <name>");
         System.out.println("    Sign in as the user with the given name.");
+        System.out.println("  uptime");
+        System.out.println("    Display the amount of time the server has been running.");
         System.out.println("  exit");
         System.out.println("    Exit the program.");
       }
     });
 
+    //UPTIME
+    //
+    // Add a command to get the server uptime when the user enters "uptime"
+    // while on the root panel.
+    //
+    panel.register("uptime", new Panel.Command() {
+      @Override
+      public void invoke(Scanner args) {
+        final ServerInfo info = context.getServerUptime();
+        if(info == null)
+          System.out.println("ERROR: unable to retrieve server uptime");
+        else {
+          long uptimeInMs = Time.now().inMs() - info.startTime.inMs();
+          long second = (uptimeInMs / 1000) % 60;
+          long minute = (uptimeInMs / (1000 * 60)) % 60;
+          long hour = (uptimeInMs / (1000 * 60 * 60)) % 24;
+
+          String formattedTime = String.format("%02d:%02d:%02d", hour, minute, second);
+          System.out.println(formattedTime);
```

Comment:
This prints when the server started, not how long it's been up.

That said, date math is a pain, and I don't see any good libraries for doing date math included in this project. So, I think for this exercise, it would be fine to just write, `"Up since " + formattedTime` and leave it at that.